src/Validation/BreadCrumbPathInterface.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ScrumWorks\OpenApiSchema\Validation;
+
+interface BreadCrumbPathInterface
+{
+    public function __toString();
+
+    public function withNextBreadCrumb(string $breadCrumb): self;
```

Comment:
bude tady `self` fungovat spravne z hlediska phpstanu? nemelo by to byt spise `static`?

Reply:
A co by melo `static` znamenat? Navic se ani do typehintu zapsat neda. Tohle znamena, ze se vraci tento interface a diky php 7.4, se ta metoda da implementovat jako, ze vraci self, tedy vraci objekt, ktery interface implementuje

Reply:
no protoze se pak muzes u navratove hodnoty spolehnout jen an to, ze je to `BreadCrumbPathInterface`, a pokud by to umelo neco navic, tak uz musis jit pres `... instanceof SomeMyBreadCrumb`, zatimco `static` (ktery zatim funguje jen jako @var typehint, ale bude pridan v PHP8.0) rika ze se vraci instance te dane tridy ktera to implementuje... o coz se da vice oprit pri staticke analyze bez nakonecnych overovani jestli je to opravdu ta a ta classa (pripadne pridavnych @var pro analyzer)